src/Tizen.NUI.Components/Controls/Extension/ButtonExtension.cs

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright(c) 2020 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+using System.ComponentModel;
+using Tizen.NUI.BaseComponents;
+
+namespace Tizen.NUI.Components
+{
+    /// <summary>
+    /// The ButtonExtension class allows developers to access the Button's components and extend their behavior in various states.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public abstract class ButtonExtension
+    {
+        /// <summary>
+        /// Called immediately after the Button creates the text part.
+        /// </summary>
+        /// <param name="button">The Button instance that the extension currently applied to.</param>
+        /// <param name="text">The created Button's text part.</param>
+        /// <param name="style">The initial style that will be appled to Button's text part.</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public virtual void OnCreateText(Button button, ref TextLabel text, TextLabelStyle style)
+        {
+        }
+
+        /// <summary>
+        /// Called immediately after the Button creates the icon part.
+        /// </summary>
+        /// <param name="button">The Button instance that the extension currently applied to.</param>
+        /// <param name="icon">The created Button's icon part.</param>
+        /// <param name="style">The initial style that will be appled to Button's icon part.</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public virtual void OnCreateIcon(Button button, ref ImageView icon, ImageViewStyle style)
```

Comment:
Can you explain why "icon" arg should be a ref?
ref must be used internally by an external reference.
But "icon" seems to always be made in here.
Can't you use "out" to here?